pymeasure/adapters/visa.py

```diff
@@ -228,7 +228,18 @@ def flush_read_buffer(self):
         in the read buffer and no END-indicator was present, read from the device until
         encountering an END indicator (which causes loss of data).
         """
-        self.connection.flush(pyvisa.constants.BufferOperation.discard_read_buffer)
+        try:
+            self.connection.flush(pyvisa.constants.BufferOperation.discard_read_buffer)
+        except NotImplementedError:
+            # fake discarding the read buffer by reading all available messages.
+            timeout = self.connection.timeout
+            self.connection.timeout = 0
+            try:
+                while True:
+                    self.read()
```

Comment:
Read bytes might be better, in case a non ASCII char is sent and the decoding in `read` would get a hickup:
```suggestion
                    self.read_bytes(-1)
```